src/ziplist.c

```diff
@@ -1527,7 +1527,7 @@ int ziplistValidateIntegrity(unsigned char *zl, size_t size, int deep,
         return 0;
 
     /* Make sure the <zltail> entry really do point to the start of the last entry. */
-    if (prev != ZIPLIST_ENTRY_TAIL(zl))
+    if (prev != NULL && prev != ZIPLIST_ENTRY_TAIL(zl))
```

Comment:
you mean that this was forgotten (or actually a bug introduced) in #9321? (doesn't seem related to #9297)

anyway, since these other PRs may (or may not) be backported to 6.2, maybe we better get that fix to a PR of it's own (so it can easily get backported too).

Reply:
ohh, on a second look, this line wasn't introduced by 9321, it always existed, but indeed it prevents 9297 from doing it's job.
i guess it also indicates that 9297 was missing some test to cover that case...
how did you discover it now?

in any way, i still prefer to take this to an independent PR.
if you're aware of other unrelated fixes in this PR that should be taken to another PR please let me know.
(only important for fixes that actually have an impact, not ones in dead code).

Reply:
Yes, due to a mistake on my part, they were to be processed in #9297.
I'll handle them in new pr.